### PR TITLE
Docker test containers

### PR DESCRIPTION
**Pull request by wilwade**
Problem
=======
We need docker containers for testing that already have the contracts loaded.

[link to Pivotal Tracker #178467941](https://www.pivotaltracker.com/story/show/178467941)

Solution
========
Build two docker containers. One Ganache based which is better for CI and one hardhat based which is better for local.

https://hub.docker.com/u/dsnp

Change summary:
---------------
* Hardhat Docker 
* Ganache Docker
* CI/CD to ship the containers to docker hub
* Rename Private Key env vars for consistent naming

Steps to Verify:
----------------
1. `docker run --rm -p 8545:8545 dsnp/ganache:latest`
2. `docker run --rm -p 8545:8545 dsnp/hardhat:latest`

### CODE REVIEW

**Review by enddynayn — APPROVED** [review]

**Review by aneyzberg — COMMENTED** [review]
lgtm

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -41,3 +41,28 @@ jobs:
 
       - name: Test
         run: npm run test
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
```
Comment (shannonwells) [inline .github/workflows/main.yml:46]: :til

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
-        node-version: [14.15.5]
     steps:
       - name: Checkout 🛎️
         uses: actions/checkout@v2
 
-      - uses: actions/setup-node@v2
+      - name: Use Node.js
+        uses: actions/setup-node@v2
         with:
+          node-version: '14.15.5'
```
Comment (shannonwells) [inline .github/workflows/release.yml:16]: This reminds me need to pick a minimum version of node and update all our repos to use it.  Try running: 
```
find . -type f -name \.tool-versions -exec grep -H nodejs {} \;
```

**Review by shannonwells — APPROVED** [review]
lgtm

File: .github/workflows/release.yml (hunk 2 above)
Comment (wilwade) [reply .github/workflows/release.yml:16]: I want to move us to node v16 for everything, but I'm not sure we can yet.

**Pull request merged**